portfolio/src/main/webapp/js/project-detail.js

```diff
@@ -40,4 +40,39 @@ for(let detail of project.detail){
      Comment.populateAll(comments);
  }
 
+
+/** Methods to hide/show new comment form and related buttons. */
+class NewCommentForm {
```

Comment:
This class is a bit weird.  With the exclusively static members and the hide and show functions only referencing the static members, this is really just a collection of functions namespaced into a class.  Consider either embracing the fact that it's a bundle of functions or else giving the class ownership over the elements that it manages.

Reply:
Agreed, this was a bit of a kludge to group some functions. In my head it was supposed to make the root level file a bit cleaner, but that didn't really work out. This has been completely refactored out in #20 .